sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsException.cs

```diff
@@ -183,7 +183,8 @@ public override string Message
         ///
         /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
         ///
-        public override string ToString() => $"{ typeof(EventHubsException).Name }({ Reason })";
+        public override string ToString() =>
+            $"{ typeof(EventHubsException).FullName }({ Reason }): { Message }{ Environment.NewLine }{ StackTrace }";
```

Comment:
Why not put the Reason info in the override for Message and just leave out overriding ToString?

Reply:
Interesting idea;  I tend to think of it as being closer to "what type of exception is this" than the description of what went wrong and, to be transparent, just  didn't think of that approach.
 
If we didn't already have the override as part of the API that was GA'd, I'd probably shift over to your suggestion.  Because we do, we'd need to keep the override with a passthrough to the base implementation to maintain binary compatibility.  Because of that, and my preference for this format, I'm going to keep this.

Reply:
Hmm would that actually be a binary break? I would think the ToString on object would make this compatible.

Reply:
Yeah, it would.   We investigated when we moved the processor to a base class.   We were asked to keep the `ToString` and other overrides on the derived type to maintain compatibility.